File: app/src/main/java/com/blog/demo/component/ParcelableValue.java
package com.blog.demo.component;

import android.os.Parcel;
import android.os.Parcelable;

public class ParcelableValue implements Parcelable {

    private int i;
    private double d;
    private String s;

    public ParcelableValue(int i, double d, String s) {
        this.i = i;
        this.d = d;
        this.s = s;
    }

    public ParcelableValue(Parcel data) {
        i = data.readInt();
        d = data.readDouble();
        s = data.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(i);
        dest.writeDouble(d);
        dest.writeString(s);
    }

    @Override
    public String toString() {
        return i + ", " + d + ", " + s;
    }

    public static final Creator<ParcelableValue> CREATOR = new Parcelable.Creator<ParcelableValue>() {

        @Override
        public ParcelableValue createFromParcel(Parcel source) {
            return new ParcelableValue(source);
        }

        @Override
        public ParcelableValue[] newArray(int size) {
            return new ParcelableValue[size];
        }
    };

}

File: app/src/main/java/com/blog/demo/component/ParcelableValue.kt
package com.blog.demo.component

import android.os.Parcel
import android.os.Parcelable

class ParcelableValue constructor(private val i: Int, private val d: Double, private val s: String?) : Parcelable {

    constructor(data: Parcel) : this(data.readInt(), data.readDouble(), data.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(i)
        parcel.writeDouble(d)
        parcel.writeString(s)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun toString(): String {
        return "$i, $d, $s"
    }

    companion object CREATOR : Parcelable.Creator<ParcelableValue> {
        override fun createFromParcel(parcel: Parcel): ParcelableValue {
            return ParcelableValue(parcel)
        }

        override fun newArray(size: Int): Array<ParcelableValue?> {
            return arrayOfNulls(size)
        }
    }


}